src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -955,6 +968,12 @@ private async Task<bool> HasSuccessfullyLoadedSlowAsync(SolutionState solution,
                 return compilationInfo.HasSuccessfullyLoaded;
             }
 
+            public async Task<ImmutableArray<SourceGeneratedDocumentState>> GetSourceGeneratedDocumentStatesAsync(SolutionState solution, CancellationToken cancellationToken)
+            {
+                var compilationInfo = await GetOrBuildCompilationInfoAsync(solution, lockGate: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+                return compilationInfo.GeneratedDocuments;
+            }
```

Comment:
i skimmed till the end of here. in general, i think we could do with more comments.